pkg/controllers/scheduling/schedule.go

```diff
@@ -145,16 +147,24 @@ func (s *pluginScheduler) Schedule(
 			return nil, err
 		}
 
-		results.scoreRecords = append(results.scoreRecords, PriorizeResult{Name: p.Name(), Scores: score})
+		// Get the prioritizer weight. Weight defined in each placement overwrites
+		// the default value defined in NewPluginScheduler().
+		weight := p.Weight()
+		if customizeWeight, err := getPlacementPrioritizerWeight(placement, p.Name()); err == nil {
```

Comment:
I think you  would want to evaluate the weight in placement at first and put it in a map at first.  So we do not need to iterate here.

Reply:
fixed.